src/csharp/Microsoft.Spark/Sql/SparkSession.cs

```diff
@@ -187,15 +187,15 @@ public void Dispose()
         /// </summary>
         /// <param name="data"><see cref="IEnumerable"/> of type <see cref="double"/></param>
         /// <returns>Dataframe object</returns>
-        public DataFrame CreateDataFrame(IEnumerable<double> data) =>
+        public DataFrame CreateDataFrame(IEnumerable<double?> data) =>
             CreateDataFrame(ToGenericRows(data), SchemaWithSingleColumn(new DoubleType()));
 
         /// <summary>
         /// Creates a Dataframe given data as <see cref="IEnumerable"/> of type <see cref="bool"/>
         /// </summary>
         /// <param name="data"><see cref="IEnumerable"/> of type <see cref="bool"/></param>
         /// <returns>Dataframe object</returns>
-        public DataFrame CreateDataFrame(IEnumerable<bool> data) =>
+        public DataFrame CreateDataFrame(IEnumerable<bool?> data) =>
```

Comment:
I don't think you can just replace this. What do you see if you run printSchema on the returned data frame? The `nullable` value should be different. If they are different (bool? => nullable = true, and bool => nullable = false), you should introduce a new set of APIs.

Reply:
Both cases return `boolean (nullable = true)` when running `df.PrintSchema()`. I have added a new set of APIs for nullable value types.

Reply:
Can you check if we can make `nullable = false` in case we pass `bool`? That's the behavior I see in Scala.

Reply:
If not, we can have just one API set. I mentioned "If they are **different** (bool? => nullable = true, and bool => nullable = false), you should introduce a new set of APIs."

Reply:
But without a new set of APIs, `var data = new List<bool>(new bool[] { true, false });` this won't work.

Reply:
Ah, it should be pretty simple to handle this case. For non-nullable types, set `StructField`'s `isNullable` to `false`.

Reply:
oh I see, let me check if we can make `nullable = false` in case we pass `bool`.

Reply:
You can just update `SchemaWithSingleColumn` to take in `isNullable = true`.

Reply:
I have added this and make `nullable = false` when passing `bool`.